RestaurantReservation.API/Program.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using RestaurantReservation;
+using RestaurantReservation.Db;
+using System.Text;
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+var tokenValidation = new TokenValidationParameters
+{
+    ValidateIssuer = true,
+    ValidateAudience = true,
+    ValidateLifetime = true,
+    ValidateIssuerSigningKey = true,
+    ValidIssuer = builder.Configuration["JWTToken:Issuer"],
+    ValidAudience = builder.Configuration["JWTToken:Audience"],
+    IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["JWTToken:Key"]))
+};
+
+builder.Services.AddSingleton(tokenValidation);
+
+
+
+
+builder.Services.AddSwaggerGen(c =>
+ {
+
+     // Add security definition for Bearer token
+     c.AddSecurityDefinition("RestaurantApiBearerAuth", new()
+     {
+         Type = SecuritySchemeType.Http,
+         Scheme = "Bearer",
+         Description = "Input a valid token to access this API"
+     });
+
+     c.AddSecurityRequirement(new()
+    {
+        {
+            new ()
+            {
+                Reference = new OpenApiReference {
+                    Type = ReferenceType.SecurityScheme,
+                    Id = "RestaurantApiBearerAuth" }
+            },
+            new List<string>()
+        }
+    });
+ });
+
+
+
+builder.Services.AddAuthorization(op =>
+{
+    op.AddPolicy("Customer", policy =>
+    {
+        policy.RequireAuthenticatedUser();
+        policy.RequireClaim("Type", "Customer");
```

Comment:
It would be better to use constant strings or enums to store such constant info.
Add this to an enum file or a constants' static class and use them where needed.